ateam_kenobi/src/play_helpers/lanes.cpp

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024 A Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "lanes.hpp"
+#include <ateam_geometry/do_intersect.hpp>
+
+namespace ateam_kenobi::play_helpers::lanes
+{
+
+ateam_geometry::Segment GetLaneLongitudinalMidSegment(const World & world, const Lane & lane)
+{
+  const auto half_field_length = world.field.field_length / 2.0;
+  const auto two_sixths_field_width = (2.0 / 6.0) * world.field.field_width;
```

Comment:
is there a reason this is 2/6 and not 1/3?

Reply:
:D I think it came from this function being derived from `GetLaneBounds` maybe? That function used 1/6 to mark the lane borders and I just made that 2/6 for the midsegments?
Kinda silly looking back at it now. I'll reduce it.